clang/test/SemaHLSL/Types/Traits/IsTypedResourceElementCompatible.hlsl

```diff
@@ -1,109 +1,28 @@
 // RUN: %clang_cc1 -triple dxil-pc-shadermodel6.6-library -finclude-default-header -fnative-half-type -verify %s
 // expected-no-diagnostics
 
-struct oneInt {

```

Comment:
Yes, it's intentional because structs shouldn't be allowed as resource element types for typed buffers. So it's redundant to test a bunch of different structs when all structs are not allowed.

Reply:
cool

Reply:
Testing at least one struct is sufficient to just ensure that structs are rejected, there's a check to ensure that the element type is not a struct. So testing different struct variations isn't very valuable.